go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -16,17 +16,8 @@ class FileParser < Dependabot::FileParsers::Base
       def parse
         dependency_set = Dependabot::FileParsers::Base::DependencySet.new
 
-        i = 0
-        chunks = module_info.lines.
-                 group_by { |line| line == "{\n" ? i += 1 : i }
-        deps = chunks.values.map { |chunk| JSON.parse(chunk.join) }
-
-        deps.each do |dep|
-          # The project itself appears in this list as "Main"
-          next if dep["Main"]
-
-          dependency = dependency_from_details(dep)
-          dependency_set << dependency if dependency
+        required_packages.each do |dep|
+          dependency_set << dependency_from_details(dep) unless dep["Indirect"]
```

Comment:
Question: is skipping `Indirect` dependencies a requirement of this change?

I get that [they're filtered out](https://github.com/dependabot/dependabot-core/blob/b07eeda901af8ef769edb678edb31647d7bc0028/go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb#L328) later, but is _that_ behaviour correct?

I don't think we should scope creep into supporting indirect dependencies in this PR - more surprised that we're not currently.

Reply:
> Question: is skipping Indirect dependencies a requirement of this change?

It's more of a consequence, really. `go list -m` only includes transitive dependencies that are included in the `go.mod` file, which apparently are only transitive dependencies that aren't go modules.

We filter indirect dependencies out much more explicitly in the [UpdateChecker](https://github.com/dependabot/dependabot-core/blob/b07eeda901af8ef769edb678edb31647d7bc0028/go_modules/lib/dependabot/go_modules/update_checker.rb#L21-L26) though, and seems to be a deliberate limitation of dependabot currently.